pkg/conf/conf.go

```diff
@@ -170,33 +174,101 @@ func GetHarvestLogPath() string {
 	return logPath
 }
 
+func GetHarvestPidPath() string {
+	var pidPath string
+	if pidPath = os.Getenv("HARVEST_PIDS"); pidPath == "" {
+		pidPath = "/var/run/harvest/"
+	}
+	return pidPath
+}
+
+func main() {
+	configFp := "/home/rahulg2/code/github/harvest/harvest.yml"
+	if Config == (HarvestConfig{}) {
+		LoadHarvestConfig(configFp)
+	}
+
+	for k, _ := range *Config.Pollers {
+		GetPrometheusExporterPorts(k, configFp)
+	}
+}
+
 /*
 GetPrometheusExporterPorts returns port configured in prometheus exporter for given poller
 If there are more than 1 exporter configured for a poller then return string will have ports as comma seperated
 */
-func GetPrometheusExporterPorts(p *node.Node, configFp string) (string, error) {
-	var port string
-	exporters := p.GetChildS("exporters")
-	if exporters != nil {
-		exportChildren := exporters.GetAllChildContentS()
-		definedExporters, err := GetExporters(configFp)
-		if err != nil {
-			return "", err
-		}
-		for _, ec := range exportChildren {
-			promNode := definedExporters.GetChildS(ec)
-			if promNode == nil {
-				fmt.Printf("poller [%s] specified exporter [%s] that does not exist\n", p.GetNameS(), ec)
-				continue
+func GetPrometheusExporterPorts(pollerName string, configFp string) (int, error) {
+	var port int
+
+	if Config == (HarvestConfig{}) {
```

Comment:
For the code to reach here the config should have already been loaded, perhaps added for debug?

Reply:
it is but given its a public method and can be consumed elsewhere. Wanted to make sure it's loaded.

Reply:
kept it as it is.